### ISSUE TITLE
Make bootstrap css and other deps from CDN available in client bundle

### ISSUE THREAD

**Issue by raisedadead** [freeCodeCamp/freeCodeCamp#35755]
We want to bundle some of our dependencies like Lato fonts, Bootstrap and MathJax with the client app.

Anything that is being pulled from CDN is not going to be available in an offline no internet environment.

Ref: #35752

**Comment by devotion-it** [comment]
Hi @raisedadead. I would like to work on this issue.

**Comment by raisedadead** [comment]
Sure, go ahead. Also you can reach out to ![Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors) for any guidance on contributing.

**Comment by pratipb** [comment]
123

**Comment by pratipb** [comment]
123

**Comment by brianamarie** [comment]
Hello @raisedadead! 👋 I've been involved in freeCodeCamp a while back when creating videos, and I recently met up with @swedishpotato at GitHub Satellite in Berlin.

@swedishpotato mentioned that he'd be really interested in moving this effort forward with some specific help on what needs to be done. @swedishpotato would you like to add more detail to why and how you'd like to help with fully offline freeCodeCamp? 

@raisedadead is the access to the CDNs offline the biggest blocker to having freeCodeCamp fully usable offline?

**Comment by raisedadead** [comment]
Hello @brianamarie 👋 !

I have been a huge fan of your work with especially on the YouTube videos and curating the quick quizzes we used to have in our early days. We have hung out before on our core team chat room :). 

On that note, I would like to extend my thanks and and gratitude in helping us hit 1M subscribers on YouTube.

Thanks for reaching out to us about the offline version of freeCodeCamp. And thanks to @swedishpotato as well for their interest in this. It's exciting to know that you guys were discussing this during GitHub Satellite.

To answer your query:

We have spent the entire last year, in re-writing the entire stack with one goal in mind, that is to make the platform into a monorepo and decouple it at the same time. It is technically possible to run freeCodeCamp **locally** (almost offline but not the whole way)  now from the `master` branch.

Here are some interesting gotchas:
- There are still some things which have had a hard dependency. Some libs, legacy artefacts, and resources as listed in the OP.
- Locally you will not need to create a user account or hit the auth service. We have a bypass that just works out of the box to save your progress. But its limited in some cases, like awarding you a certificate of completion, or validating a project, because of obvious server side validation reasons.
- Locally it takes nearly ~60 mins to build the entirety of freeCodeCamp's English language open source community content. ~10K Guide articles and ~2K Coding Challenges, which is ever increasing.

This makes it somewhat technical for an average user to use freeCodeCamp locally as of now.
There probably a bit more things that I am missing out right now, but we are just weeks away from being a able to push this version of freeCodeCamp to production.

There is also a overhaul of the UX planned later after we go to production, which will further reduce the dependency on more of these resources.

**Please note that this will give some in browser offline abilities like a PWA.** Because this new version is aimed at being a JAMStack app.

So in short we are getting closer.

**To achieve a truly offline (No-Internet/In-Flight) mode:**

- We want zero dependency on anything that's used from cdns.
- We want a electron or similar environment to be able to run our nodejs binaries for the back-end and serve the react front-end.
- We want to use a lighter NoSQL, database than MongoDB only for the offline version.
- We also want to build on a seamless release pipeline to achieve building this for all types of platforms.  

So you can see that we are there half way or at least have the backbone ready for 'true' offline mode. But there is some work to be done. This involves a some more considerations on the architectural side of things as well.

We are getting there, and its only a matter of time :)

And of course it goes, without saying we would love all the help that we can get.

Thanks for connecting, hope this helps.

**Comment by swedishpotato** [comment]
Hi @raisedadead - this is really excellent news! A couple of follow up questions if I may (I am only a below average developer, so please bear with me!).

Q1. Are there any instructions currently on how to get a (even somewhat) offline working version up and running?

Q2. Is there anything we can do to help? Is there any "grunt work" that we could do for instance?

Please keep us informed of the offline version launch! I think this is a MASSIVE thing, because not only do prisoners learning to code have no access to the internet, around **half of the world's current population have no (or no reliable) access to the internet**, but this should not stop them from learning to code! I imagine a FreeCodeCamp USB stick that can be plugged into more of less any old laptop which then allows people to study code via the FreeCodeCamp curriculum. Awesome!

**Comment by raisedadead** [comment]
Yes, our [Guidelines for Contributing](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md), explain all the details, [specifically this doc here](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-setup-freecodecamp-locally.md) is what you are most likely interested in.

Then on you could turn off your internet (or use offline mode in devtools of your browser to simulate), see what breaks and propose solutions.

We are quite swamped at the minute to do this, but eventually we will get there where you can install and run freeCodeCamp off of portable media and fulfil the dreams of being fully offline.

**Comment by raisedadead** [comment]
For example, to answer your second query, most css would break because of a really old bootstrap dependency. And MathJax and things like that.
